main.go

```diff
@@ -152,3 +172,45 @@ func hasLabel(labels map[string]string, labelName string) bool {
 	_, ok := labels[labelName]
 	return ok
 }
+
+func hasTaint(taints []v1.Taint, taintKey string) bool {
+	for _, taint := range taints {
+		if taint.Key == taintKey {
+			return true
+		}
+	}
+	return false
+}
+
+func isControlPlaneNode(node v1.Node, ctrlClient client.Client) bool {
+	if hasLabel(node.Labels, MasterNodeRoleLabel) || hasLabel(node.Labels, ControlPlaneNodeRoleLabel) || hasLabel(node.Labels, LegacyMasterNodeLabel) {
+		return true
+	}
+
+	// if node doesn't have any of the control plane labels, check if it has control plane Pods running as fallback.
+	// During DR scenarios, the node may not have any of the control plane labels but may still be a control plane node.
+	var apiPods v1.PodList
+	err := ctrlClient.List(context.TODO(), &apiPods, client.InNamespace("kube-system"), client.MatchingLabels{"component": "kube-apiserver", "tier": "control-plane"})
```

Comment:
I think it would make sense to reuse the context from main

Reply:
right ofc